tensorpipe/test/channel/channel_test.h

```diff
@@ -202,6 +203,62 @@ class ChannelTestCase {
 
 template <typename TBuffer>
 class ClientServerChannelTestCase : public ChannelTestCase<TBuffer> {
+  std::vector<std::future<std::shared_ptr<tensorpipe::transport::Connection>>>
+  accept(
+      std::shared_ptr<tensorpipe::transport::Listener> listener,
+      size_t numConnections) {
+    auto promises = std::make_shared<std::vector<
+        std::promise<std::shared_ptr<tensorpipe::transport::Connection>>>>(
+        numConnections);
+    for (size_t i = 0; i < numConnections; ++i) {
+      listener->accept(
+          [&, promises](
+              const tensorpipe::Error& error,
+              std::shared_ptr<tensorpipe::transport::Connection> connection) {
+            ASSERT_FALSE(error) << error.what();
+            connection->read([promises, connection](
```

Comment:
Nit: you could move `promises` into the lambda.

Reply:
The `promises` vector needs to be shared among all lambdas. Am I missing something?

Reply:
A copy of the shared_ptr is already made by the closure of the first lambda (line 215), hence this shared_ptr here will not need to be copied further, and that's why it can be moved.